### ISSUE TITLE
Ask before making shared resources like animations or collision shapes

### ISSUE THREAD

**Issue by ScyDev** [godotengine/godot#3104]
Silently making resources like animations, collision shapes shared among scenes can be very ugly and time consuming to fix. It can mess up a large portion of the project.

Example 1: I build many levels. Each new level I copy from another, existing one because they are complex and it would be unnecessary work to build each level from the ground up.
All collision shapes are now shared among all levels and if one is changed, they all change and mess up the levels.

Example 2: Same as nr. 1, but with powerups or enemies. Very easy to make that mistake and mess up all the enemies without noticing it.

Example 3: Making animations for enemies. When saving a copy of an enemy to create the next one. Adjust the frames on the animation (spritesheet) and now all the enemies made before have messed up animations.

TLDR: Sharing these resources during copy should be a special function and not the default. In most cases you want unique resources.

**Comment by reduz** [comment]
I don't understand this.. Resources are not shared between scenes until you
explicitly save them to disk

**Comment by ScyDev** [comment]
Aren't all resources saved to disk?

Easy to test:
Create a static body and add a collision shape (box). Save that scene. Then save that scene again under a different name. Then modify the box collision shape. Then open the first scene and see that the collision shape is changed there too.

**Comment by Ovnuniarchos** [comment]
What's your version/commit? I did your steps and the two scenes are different.

**Comment by reduz** [comment]
I think the issue is mostly an editor bug/feature and not really related to
in-game. Will have to check it out,

**Comment by ScyDev** [comment]
Yes, this behaviour is while editing, not while the game is running.

**Comment by bojidar-bg** [comment]
Bump. Is this still valid, and what solutions can we come up with?

**Comment by reduz** [comment]
Its perfect that resources are shared, but we sould probably add a
non-share duplicate shortcut like shift-ctrl-d (with corresponding dropdown
menu item). Maybe as the OP says, we coukd add a warning the first time you
duplicate animation or collisionobject (one for each)... So at least user
knows this is the default behavior.

**Comment by akien-mga** [comment]
Feature and improvement proposals for the Godot Engine are now being discussed and reviewed in a dedicated [**Godot Improvement Proposals (GIP)**](https://github.com/godotengine/godot-proposals) ([godotengine/godot-proposals](https://github.com/godotengine/godot-proposals)) issue tracker. The GIP tracker has a detailed issue template designed so that proposals include all the relevant information to start a productive discussion and help the community assess the validity of the proposal for the engine.

The main ([godotengine/godot](https://github.com/godotengine/godot)) tracker is now solely dedicated to bug reports and Pull Requests, enabling contributors to have a better focus on bug fixing work. Therefore, we are now closing all older feature proposals on the main issue tracker.

If you are interested in this feature proposal, please [**open a new proposal on the GIP tracker**](https://github.com/godotengine/godot-proposals/issues/new/choose) following the given issue template (after checking that it doesn't exist already). Be sure to reference this closed issue if it includes any relevant discussion (which you are also encouraged to summarize in the new proposal). Thanks in advance!
